core/src/main/java/com/flipkart/varadhi/core/cluster/MessageHandler.java

```diff
@@ -0,0 +1,12 @@
+package com.flipkart.varadhi.core.cluster;
+
+import java.util.concurrent.CompletableFuture;
+
+public interface MessageHandler {
+
+    // handle will be used for messages sent via channel.send() and channel.publish().
+    <E extends ClusterMessage> CompletableFuture<Void> handle(E message);
+
+    // request will be used for messages sent via channel.request().
+    <E extends ClusterMessage> CompletableFuture<ResponseMessage> request(E message);
```

Comment:
```suggestion
    <E extends ClusterMessage> CompletableFuture<ResponseMessage> handleRequest(E message);
```

Does the params need sender info? E message might not have sender info.